### DISCUSSION TITLE
ImportError: DLL load failed while importing _solver: The specified module could not be found.

### DISCUSSION THREAD

**Discussion by cylin928** [pyswmm/pyswmm#323] [Q&A] [answered]
I am using python 3.9.7 through spyder 5.1.5.
I successfully installed the pyswmm. However, when I try to import it by
"from pyswmm import Simulation"
I got the following errors

from pyswmm import Simulation
Traceback (most recent call last):

  File "C:\Users\AppData\Local\Temp/ipykernel_12088/3138147840.py", line 1, in <module>
    from pyswmm import Simulation

  File "C:\Users\anaconda3\envs\cyber\lib\site-packages\pyswmm\__init__.py", line 11, in <module>
    from pyswmm.links import Link, Links

  File "C:\Users\anaconda3\envs\cyber\lib\site-packages\pyswmm\links.py", line 12, in <module>
    from pyswmm.swmm5 import PYSWMMException

  File "C:\Users\anaconda3\envs\cyber\lib\site-packages\pyswmm\swmm5.py", line 20, in <module>
    from swmm.toolkit import solver

  File "C:\Users\anaconda3\envs\cyber\lib\site-packages\swmm\toolkit\solver.py", line 18, in <module>
    from . import _solver

ImportError: DLL load failed while importing _solver: The specified module could not be found.

Please help me figure this out.

**Comment by jennwuu** [comment]
Hey @philip928lin, 

Thanks for using pyswmm. This error was identified in swmm-python when using pyswmm and anaconda: https://github.com/OpenWaterAnalytics/swmm-python/issues/79 where solver cannot be found. This should be addressed in the next release.

In the meantime, can you try pyswmm with regular Python?

Thanks,
Jenn

CC: @michaeltryby - I think this is related to the issue identified in swmm-python, which was patched in a PR.

**Comment by michaeltryby** [comment]
@philip928lin sorry you are having problems. What version of pyswmm are you using?

**Comment by cylin928** [comment]
I use this version
pyswmm                    1.1.0.dev0               pypi_0    pypi
swmm-toolkit              0.8.2                    pypi_0    pypi

Let me know if there is any way that I can manually add environment paths to fix this issue. 
For example what paths are needed.
Thank you a lot for your prompt reply.

**Comment by michaeltryby** [comment]
@philip928lin if you set this environment variable in the shell prior to launching Python and the wheel should work properly.

CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1

**Comment by cylin928** [comment, marked as answer]
Thank you. 
The code can run smoothly if adding
`import os`
`os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE']="1" `
at the top of the script.

**Comment by zhaoguohan2021** [comment]
I also run into the same problem, since I install the pyswmm in anaconda env. 
The solution works for me is to go to this directory below:
C:\Users\QM03JZ\Anaconda3\envs\pyswmm\Lib\site-packages\swmm\toolkit

and **edit _init_.py** by adding **os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE']="1"**

if platform.system() == "Windows":
    libdir = os.path.join(os.path.dirname(__file__), "../../swmm_toolkit")

    os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE']="1"
    if hasattr(os, 'add_dll_directory'):    
        os.add_dll_directory(libdir)
    else:
        os.environ["PATH"] = libdir + ";" + os.environ["PATH"]
